File: Java/StrategyPattern/src/Duck/Duck.java
package Duck;

import DuckBehaviours.Fly.IFlyBehaviour;
import DuckBehaviours.Quack.IQuackBehaviour;

public abstract class Duck {
    protected IFlyBehaviour flyBehaviour;

    protected IQuackBehaviour quackBehaviour;

    public Duck(IFlyBehaviour flyBehaviour, IQuackBehaviour quackBehaviour) {
        this.flyBehaviour = flyBehaviour;
        this.quackBehaviour = quackBehaviour;
    }

    public void performFly(){
    }

    public void performQuack(){
    }
}

File: Kotlin/StrategyPattern/src/ducks/Duck.kt
package ducks

import duckBehaviours.flyingBehaviour.IFlyingBehaviour
import duckBehaviours.quackBehaviour.IQuackBehaviour

abstract class Duck(var iFlyingBehaviour: IFlyingBehaviour, var iQuackBehaviour: IQuackBehaviour) {

    abstract fun performFly()

    abstract fun performQuack()
}